### ISSUE TITLE
lookup is not defined

### ISSUE THREAD

**Issue by Sendoushi** [maxkueng/node-musicbrainz#23]
I'm using this under an API.
Everything works fine but sometimes I get:

``` json
{
  "code": "InternalError",
  "message": "lookup is not defined"
}
```

I can't event catch the error.

**Comment by maxkueng** [comment]
Weird. Looks like this is a response from the MusicBrainz API.

I've been meaning to rewrite this library once their JSON API becomes stable but it's been in beta for years.

**Comment by Sendoushi** [comment]
Yes. For me, when I did the pr, It was a little confusing to understand the library.
If you want... I can help figuring out a way to rewrite the library and then we could proceed with it under another branch.

**Comment by richdunajewski** [comment]
The error is in: lib\musicbrainz.js:909

```javascript
// If the service is busy, we'll try again later
if (response.statusCode == 503) {
    timers.setTimeout(lookup, 2000, callback);
    return;
}
```

Should this be `this._lookup`? Not sure what it should point to. There's another line, line 824 that has the same call.

So this happens when the MusicBrainz API returns a 503 for being busy. If you're testing with the public API, you'll get a lot of these. The library is trying to reschedule a request in 2 seconds but the function is undefined.

**Comment by krahimian** [comment]
it should be ```search```

**Comment by maxkueng** [comment]
I've published v0.2.7 which should fix this
